akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/DispatcherSelectorSpec.scala

```diff
@@ -8,119 +8,156 @@ import com.typesafe.config.Config
-      pingPong ! Ping(probe.ref)
+    "default to unbounded" in {
+      val actor = spawn(behavior)
+      val (_, mailbox) = actor.ask(WhatsYourDispatcherAndMailbox).futureValue
+      mailbox shouldBe a[NodeMessageQueue]
+    }
+
+    "select a blocking dispatcher" in {
+      val actor = spawn(behavior, DispatcherSelector.blocking())
+      val (dispatcherId, _) = actor.ask(WhatsYourDispatcherAndMailbox).futureValue
+      dispatcherId shouldBe Dispatchers.DefaultBlockingDispatcherId
+    }
 
-      val response = probe.receiveMessage()
-      response.threadName should startWith("DispatcherSelectorSpec-ping-pong-dispatcher")
+    "select dispatcher from config" in {
+      val actor = spawn(behavior, Props.empty.withDispatcherFromConfig("specific-dispatcher"))
+      val (dispatcherId, _) = actor.ask(WhatsYourDispatcherAndMailbox).futureValue
+      dispatcherId shouldBe "specific-dispatcher"
     }
 
     "detect unknown dispatcher from config" in {
-      val probe = createTestProbe[Pong]()
+      val probe = createTestProbe[(String, MessageQueue)]()
       LoggingTestKit.error("Spawn failed").expect {
-        val ref = spawn(PingPong(), Props.empty.withDispatcherFromConfig("unknown"))
+        val ref = spawn(behavior, Props.empty.withDispatcherFromConfig("unknown"))
         probe.expectTerminated(ref)
       }
     }
 
     "select same dispatcher as parent" in {
-      val parent = spawn(SpawnProtocol(), Props.empty.withDispatcherFromConfig("ping-pong-dispatcher"))
-      val childProbe = createTestProbe[ActorRef[Ping]]()
-      parent ! SpawnProtocol.Spawn(PingPong(), "child", Props.empty.withDispatcherSameAsParent, childProbe.ref)
+      val parent = spawn(SpawnProtocol(), Props.empty.withDispatcherFromConfig("specific-dispatcher"))
 
-      val probe = createTestProbe[Pong]()
-      val child = childProbe.receiveMessage()
-      child ! Ping(probe.ref)
+      val child = parent.ask { replyTo: ActorRef[ActorRef[WhatsYourDispatcherAndMailbox]] =>
+        SpawnProtocol.Spawn(behavior, "child", DispatcherSelector.sameAsParent(), replyTo)
+      }.futureValue
 
-      val response = probe.receiveMessage()
-      response.threadName should startWith("DispatcherSelectorSpec-ping-pong-dispatcher")
+      val (dispatcherId, _) = child.ask(WhatsYourDispatcherAndMailbox).futureValue
+      dispatcherId shouldBe "specific-dispatcher"
     }
 
     "select same dispatcher as parent, several levels" in {
-      val grandParent = spawn(SpawnProtocol(), Props.empty.withDispatcherFromConfig("ping-pong-dispatcher"))
-      val parentProbe = createTestProbe[ActorRef[SpawnProtocol.Spawn[Ping]]]()
-      grandParent ! SpawnProtocol.Spawn(
-        SpawnProtocol(),
-        "parent",
-        Props.empty.withDispatcherSameAsParent,
-        parentProbe.ref)
-
-      val childProbe = createTestProbe[ActorRef[Ping]]()
-      grandParent ! SpawnProtocol.Spawn(PingPong(), "child", Props.empty.withDispatcherSameAsParent, childProbe.ref)
```

Comment:
It seems that the 'parent' and 'child' do not spawn with a vertical hierarchy, but rather a horizontal hierarchy.

Reply:
yes, seems to be a mistake, it's spawning from grandParent twice